src/main/java/seedu/address/model/customer/Customer.java

```diff
@@ -44,8 +44,16 @@ public Customer(CustomerBuilder builder) {
         phone = builder.phone;
         email = builder.email;
         tags = builder.tags;
-        commissions = builder.commissions;
         address = builder.address;
+        commissions = new UniqueCommissionList();
+        builder.commissions.forEach(commission -> {
+            commissions.add(new Commission.CommissionBuilder(
+                    commission.getTitle(),
+                    commission.getFee(),
+                    commission.getDeadline(),
+                    commission.getCompletionStatus(),
+                    commission.getTags()).setIterations(commission.getIterations()).build(this));
```

Comment:
I noticed that the Commissions created by `CommissionBuilder.build` references the same iteration list as the original customer's commissions, perhaps we can do something similar to this change in the `Commission` constructor as well so the new commission's iterations will be different as well.

Reply:
Just curious, why do we need to make a copy of the iterations as well? Since the iterations don't store a reference to a commission. Or is this sort of just defensive programming because each iteration can only be associated to one commission?

Reply:
Yes, it's mainly for defensive programming, since it might be possible that we create a CommisionBuilder `cb` and set the iterations to those of commissionX, and when we run `commissionY = cb.build()`, both `commissionX` and `commissionY` would have the same iterations list and adding to commissionX's list will affect commissionY as well. This might not be a problem for us right now, but I can't be entirely sure without checking all uses of `CommissionBuilder::build`